src/devices/cardin.c

```diff
@@ -92,40 +102,56 @@ static int cardin_callback(bitbuffer_t *bitbuffer) {
 				dip[8]='+';
 		}
 
-		fprintf(stdout, "                 123456789\n");
-		fprintf(stdout, "dipswitch      = %s\n\n",dip);
-
-		fprintf(stdout, "                 -->ON\n");
-		fprintf(stdout, "right button   = ");
+		// Jumpers for the right button
 		if((bb[0][2] & 3) == 3) {
-			fprintf(stdout,                  "2 --o (this is right button)\n");
-			fprintf(stdout, "                 1 --o\n");
+			strcpy(rbuttonJ2, "--o (this is right button)");
+			strcpy(rbuttonJ1, "--o");
 		}
 		if((bb[0][2] & 9) == 9) {
-			fprintf(stdout,                  "2 --o (this is right button)\n");
-			fprintf(stdout, "                 1 o--\n");
+			strcpy(rbuttonJ2, "--o (this is right button)");
+			strcpy(rbuttonJ1, "o--");
 		}
 		if((bb[0][2] & 12) == 12) {
-			fprintf(stdout,                  "2 o-- (this is left button or two buttons on same channel)\n");
-			fprintf(stdout, "                 1 o--\n");
+			strcpy(rbuttonJ2, "o-- (this is left button or two buttons on same channel)");
```

Comment:
currently your output is either of
```
"--o (this is right button)" "--o"
"--o (this is right button)" "o--"
"o-- (this is left button or two buttons on same channel)" "o--"
"o-- (this is right button)" "--o"
```
it could just be ("dip2" not "rbuttonJ2" and "rbuttonJ1")
```
"--o RB --o"
"--o RB o--"
"o-- LB/2B o--"
"o-- RB --o"
```
Anyway, I don't get that "--o" symbol. I thought those DIPs are tristate with -=off +=on o=float?
But it seems to be ascii-art for a single DIP slider? If so, that's a good picture for debug output but structured output then should be:
```
"00"
"01"
"11?"
"10"
```
If those DIPs are not tristate then a bitfield (`010101010`) would be much cleaner that the -o+ symbols for the "dipswitch".

You can keep the ascii-art output for debug mode by the way.